capstone/src/main/webapp/script.js

```diff
@@ -6,3 +6,48 @@ var instance = M.Tabs.init(elem, options);
 //Initialize sidenav
 const slide_menu = document.querySelectorAll(".sidenav");
 M.Sidenav.init(slide_menu, {});
+
+//Get Search Request From User and display result
+async function studySetsBySearch() {
+  const stringToSearchBy = document.getElementById("search").value;
+  const response = await fetch(
+    "/study_set?stringToSearchBy=" + stringToSearchBy
+  );
+  const searchResponse = await response.json();
+  var configuredSetsToPost = configureCardsForSearchPage(searchResponse);
+  document.getElementById("results-container").innerHTML = configuredSetsToPost;
+}
+
+function configureCardsForSearchPage(studySets) {
+  var configuredCardStudySets = "";
+  var counter = 0;
+
+  for (var studySet of studySets) {
+    configuredCardStudySets += configureCard(studySet);
+    counter++;
+  }
+
+  if (counter == 0) {
```

Comment:
Move this `if` up to the top of the function and use `studySets.length` instead of a `counter` variable